x-pack/plugins/fleet/server/services/epm/packages/install.ts

```diff
@@ -247,32 +190,75 @@ async function installPackageFromRegistry({
   pkgkey,
   esClient,
   force = false,
-}: InstallRegistryPackageParams): Promise<AssetReference[]> {
+}: InstallRegistryPackageParams): Promise<InstallResult> {
+  const logger = appContextService.getLogger();
   // TODO: change epm API to /packageName/version so we don't need to do this
   const { pkgName, pkgVersion } = Registry.splitPkgKey(pkgkey);
+
   // get the currently installed package
   const installedPkg = await getInstallationObject({ savedObjectsClient, pkgName });
   const installType = getInstallType({ pkgVersion, installedPkg });
-  // let the user install if using the force flag or needing to reinstall or install a previous version due to failed update
-  const installOutOfDateVersionOk =
-    installType === 'reinstall' || installType === 'reupdate' || installType === 'rollback';
 
+  // get latest package version
   const latestPackage = await Registry.fetchFindLatestPackage(pkgName);
-  if (semverLt(pkgVersion, latestPackage.version) && !force && !installOutOfDateVersionOk) {
-    throw new PackageOutdatedError(`${pkgkey} is out-of-date and cannot be installed or updated`);
+
+  const forceInstall = force || ['reinstall', 'reupdate', 'rollback'].includes(installType);
```

Comment:
I think this is going to cause a reinstall of a package when the `_bulk` API is hit.

If the currently installed version is `1.0.0` I think `installType` will return `reinstall` which would result in `forceInstall` being `true`. The `_bulk` API is only passed a package name and it then requests the latest version. If the user already has the latest version installed would this reinstall it?

I think I noticed this in my testing because I saw logs for the transform being started and stopped each time the `_bulk` api is hit.

The `upgradePackage` function avoiding reinstalling a package when the current version equaled the latest version.

This could be problematic because a request to the `_bulk` API is made pretty frequently from the security solution app.

Reply:
thanks for spotting this and testing! I noticed this too when I was testing, it's a bug I introduced late yesterday 😅  I will fix in my next update

Reply:
pushed update that should fix this